File: w07_d4_stutter/stutter.js
console.log('Stutter');

// accum('abcd') - - >> "A-Bb-Ccc-Dddd"

// split characters
// make one of 1st, 2 of 2nd, etc
// capitalize first
// lower subsequent
// join with '-'
// return the lot

// bonus: accept a second argument, 'step'

function accum(word) {
  if(word === word) {
    console.log(word);
  }

  var split = word.split('');

  for (var i = 0; i < length; i++) {
    word += ([split[i]]);
  }
  console.log(word);
  return accum;
}

// harrys solution

var accum = function(word, step) {
    step = step || 1;

    return word.toLowerCase().split('').map(function(e, i) {
        var stutt = e.repeat(i + step);
        return stutt.charAt(0).toUpperCase() + stutt.slice(1);
    }).join('-');
};

console.log(accum('cwAt'));

File: w07_d4_stutter/stutter.rb
# jordans solution

def accum(string, step=1)
  string_splited = string.downcase.split('')
  final_arr = []

  # FIRST LOOP TO LOOP THROUGH EACH LETTER
  string_splited.each_with_index do |letter, index|
    i = 0
    temp_arr = []
    # SECOND LOOP TO LOOP AND MULTIPLY EACH LETTER
    while i <= index + step do
      # IF FIRST LETTER ACCORDING TO INDEX, UPCASE
      if i == 0
        temp_arr << letter.upcase
      else
        temp_arr << letter
      end
      i += 1
    end
    final_arr << temp_arr.join('')
  end
  puts final_arr.join('-')
end

accum("abcd")
    # --> "A-Bb-Ccc-Dddd"
accum("RqaEzty")
    # --> "R-Qq-Aaa-Eeee-Zzzzz-Tttttt-Yyyyyyy"
accum("cwAt")
    # --> "C-Ww-Aaa-Tttt"






# deans solution

// accum("abcd") --> "A-Bb-Ccc-Dddd"
// accum("RqaEzty") --> "R-Qq-Aaa-Eeee-Zzzzz-Tttttt-Yyyyyyy"
// accum("cwAt") --> "C-Ww-Aaa-Tttt"

// Returns a random number between min (inclusive) and max (exclusive)
function getRandomArbitrary(min, max) {
  return Math.random() * (max - min) + min;
}

function capitalize(substring) {
    return substring.charAt(0).toUpperCase() + substring.slice(1);
}

function accum(inputString, step) {
  splitString = inputString.split("");
  finalString = "";

  splitString.forEach( function(l) {
    // console.log(l);
    if (finalString != "") {
        finalString += '-';
    }
    rand = getRandomArbitrary(1,8);

    subString = ""
    for (var i = 1;i < rand;i++) {
      subString += l.toLowerCase();
    }
    finalString += capitalize(subString);
  })
return finalString;
}
console.log(accum("abcd", 2));
console.log(accum("RqaEzty", 2));
console.log(accum("cwAt", 2));
